sympy/functions/elementary/integers.py

```diff
@@ -144,6 +144,12 @@ def _eval_as_leading_term(self, x, logx=None, cdir=0):
         arg = self.args[0]
         arg0 = arg.subs(x, 0)
         r = self.subs(x, 0)
+        if arg0 is S.NaN:
+            limit = arg.limit(x, 0, dir='-' if re(cdir).is_negative else '+')
+            ndir = arg.dir(x, cdir=cdir)
+            if not limit.is_integer:
```

Comment:
`is.integer` does not work with floating point numbers. There is no guarantee that the limit would never be a `Float`. It seems that the test used in the next block, `arg0 == r`, would be more reliable. For that purpose also `r` should be redefined, probably as `floor(arg0)` (where `arg0` is the limit). There is no need to repeat the code here. We could just let the control pass to the next block after `arg0` and `r` have been redefined.

Reply:
Good catch this , had completely overlooked this and didn't cross my mind even once.
> There is no guarantee that the limit would never be a Float

I also get what you're trying to convey . The redefining approach will also make it easier to debug or catch errors in the future .Will make the changes. Thanks !

Reply:
Resolving as addressed !